### PR TITLE
Add neovim recipe

### PR DESCRIPTION
**Pull request by curtisalexander**

### CODE REVIEW

**Comment by jakirkham** [comment]
This should be renamed to `python-neovim`. Others are working on packaging the `neovim` editor in `conda` as well. ( https://github.com/conda-forge/staged-recipes/pull/1677 ).

File: recipes/python-neovim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - msgpack-python >=0.4.0
+    - greenlet
+    - trollius  # [py27]
```
Comment (jakirkham) [inline recipes/python-neovim/meta.yaml]: Do we need all these at build time? Guessing we only need `python` and `setuptools`. Can we please update the listing in `build` accordingly?

File: recipes/python-neovim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - neovim
+    - neovim.api
+    - neovim.msgpack_rpc
+    - neovim.msgpack_rpc.event_loop
+    - neovim.plugin
+
+about:
+  home: http://github.com/neovim/python-client
+  license: Apache
+  license_family: Apache
```
Comment (jakirkham) [inline recipes/python-neovim/meta.yaml]: Please add `license_file: LICENSE`.

File: recipes/python-neovim/meta.yaml (hunk 1 above)
Comment (curtisalexander) [inline recipes/python-neovim/meta.yaml]: No, I don't think so at at build time.  I have been confusing and mixing up those things needed at build time versus those that are needed at runtime.

File: recipes/python-neovim/meta.yaml (hunk 2 above)
Comment (curtisalexander) [inline recipes/python-neovim/meta.yaml]: Ah, forgot.

File: recipes/python-neovim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - neovim.api
+    - neovim.msgpack_rpc
+    - neovim.msgpack_rpc.event_loop
+    - neovim.plugin
+
+about:
+  home: http://github.com/neovim/python-client
+  license: Apache-2.0
+  license_family: Apache
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (bgruening) [inline recipes/python-neovim/meta.yaml]: `license_file: 'LICENSE'` should be enough isn't it?

Comment (curtisalexander) [inline recipes/python-neovim/meta.yaml]: Not certain, I'll try and resubmit to rebuild.

Comment (curtisalexander) [inline recipes/python-neovim/meta.yaml]: Looks like the `license_file` section can be simplified.  Made the edit within 43bc154 all built successfully.

**Comment by jakirkham** [comment]
It looks like the license file from staged-recipes got moved into this recipe. Could you please revert that? Also please squash all commits afterwards.

**Comment by curtisalexander** [comment]
@jakirkham Hmm...I thought I squashed the commits but I may not have done so correctly.  If I didn't, then please let me know and I'll try to dig myself out of the `git` hole I've dug.  I'm definitely an occasional `git` user.

I did, however, remove the `LICENSE` file.  It was never needed within the directory, I don't know what I was thinking.

**Comment by jakirkham** [comment]
No worries. I tried to squash this in PR ( https://github.com/conda-forge/staged-recipes/pull/2914 ). Could you please take a look and verify that it is ok?

File: LICENSE [hunk 4]
```diff
@@ -1,27 +0,0 @@
-Copyright (c) 2015-2017, conda-forge
-All rights reserved.
```
Comment (jakirkham) [inline LICENSE]: FYI this is the license I was referring to as having been moved. We do want to make sure to keep this one as it is the license for staged-recipes as a whole. Made sure it is kept in the rebased PR.

Comment (curtisalexander) [inline LICENSE]: I understand now what you were referring to. Yes, I was moving files around and should not have deleted the `LICENSE` file from the `staged-recipes` directory. Are you planning to just close this PR and let [#2914](https://github.com/conda-forge/staged-recipes/pull/2914) replace?

Thanks.

Comment (jakirkham) [inline LICENSE]: Yeah, the plan is to replace this PR with that one. Just wanted you to be aware of this issue for future submissions.

**Comment by curtisalexander** [comment]
Noted - thank you for the info.

**Pull request closed without merging**